### PR TITLE
TOF fix

### PR DESCRIPTION
**Pull request by iVoider**
Don't forget to add libboringssl.tbd and libcoretls_cfhelpers.tbd

### CODE REVIEW

**Comment by PencilNavigator** [comment]
“StolenFromMe” LOL

**Comment by IsaacMarovitz** [comment]
@iVoider you need to document the full process of defining `CS_OPS_STATUS` and `CS_OPS_IDENTITY`. Also please describe what you mean by `Don't forget to add libboringssl.tbd and libcoretls_cfhelpers.tbd`

**Comment by TheMoonThatRises** [comment]
I think for `libboringssl.tbd` and `libcoretls_cfhelpers.tbd` you need to add it to frameworks and libraries in PlayTool's target general settings. As for `CS_OPS_STATUS` and `CS_OPS_IDENTITY`, I could only find references to `CS_OPS_STATUS` https://opensource.apple.com/source/xnu/xnu-2782.1.97/bsd/sys/codesign.h referencing codesigning.

**Comment by IsaacMarovitz** [comment]
@iVoider I can't get this to work. I can see serves, but I can't join them. The same graphical issues as before are present.

File: PlayTools/PlayLoader.m [hunk 1]
```diff
@@ -90,6 +90,139 @@ static int pt_sysctlbyname(const char *name, void *oldp, size_t *oldlenp, void *
 DYLD_INTERPOSE(pt_uname, uname)
 DYLD_INTERPOSE(pt_sysctlbyname, sysctlbyname)
 DYLD_INTERPOSE(pt_sysctl, sysctl)
+    
+    struct dyld_interpose_tuple {
```
Comment (IsaacMarovitz) [inline PlayTools/PlayLoader.m:94]: Fix indentation

File: PlayTools/PlayLoader.m [hunk 2]
```diff
@@ -90,6 +90,139 @@ static int pt_sysctlbyname(const char *name, void *oldp, size_t *oldlenp, void *
...
+extern void dyld_dynamic_interpose(const struct mach_header* mh, const struct dyld_interpose_tuple array[], size_t count);
+
+extern int csops(pid_t pid, unsigned int ops, void *useraddr, size_t usersize);
+
+extern Boolean SecKeyVerifySignature(SecKeyRef key, SecKeyAlgorithm algorithm, CFDataRef signedData, CFDataRef signature, CFErrorRef  _Nullable *error);
+
+extern OSStatus SecTrustEvaluate(SecTrustRef trust, SecTrustResultType *result);
+
+int my_csops(pid_t pid, uint32_t ops, user_addr_t useraddr, user_size_t usersize){
+    if (ops == CS_OPS_STATUS || ops == CS_OPS_IDENTITY) {
```
Comment (IsaacMarovitz) [inline PlayTools/PlayLoader.m:109]: You need to include defines for these two variables

File: PlayTools/PlayLoader.m [hunk 3]
```diff
@@ -90,6 +90,139 @@ static int pt_sysctlbyname(const char *name, void *oldp, size_t *oldlenp, void *
...
+}
+
+extern void SSL_set_custom_verify(void *ssl, int mode, int (*callback)(void *ssl, uint8_t *out_alert));
+
+void my_SSL_set_custom_verify(void *ssl, int mode, int (*callback)(void *ssl, uint8_t *out_alert)){
+    SSL_set_custom_verify(ssl, SSL_VERIFY_NONE, custom_verify_callback_that_does_not_validate);
+    return;
+}
+
+DYLD_INTERPOSE(my_SSL_set_custom_verify, SSL_set_custom_verify)
```
Comment (IsaacMarovitz) [inline PlayTools/PlayLoader.m:207]: Please group all the function implementations and interposes together styled in the same way the ones currently in PlayLoader are styled.

File: PlayTools/PlayLoader.m [hunk 4]
```diff
@@ -90,6 +90,139 @@ static int pt_sysctlbyname(const char *name, void *oldp, size_t *oldlenp, void *
...
+    return "Apple";
+}
+
+enum ssl_verify_result_t {
+    ssl_verify_ok = 0,
+    ssl_verify_invalid,
+    ssl_verify_retry,
+};
+
+#define SSL_VERIFY_NONE 0
```
Comment (IsaacMarovitz) [inline PlayTools/PlayLoader.m:192]: Move this define to the top with the other defines

**Review by IsaacMarovitz — CHANGES_REQUESTED** [review]

**Comment by IsaacMarovitz** [comment]
> Don't forget to add libboringssl.tbd and libcoretls_cfhelpers.tbd

If those files are required include the change to the Xcode Project in your PR.

**Pull request closed without merging**